pkg/infra/game/state/state.go

```diff
@@ -60,13 +60,19 @@ func (s *AgentState) ChangeShieldInUse(shieldIdx decision.ItemIdx) {
 	}
 }
 
+type PotionSlice struct {
+	HPpotion []uint //index is potion's id and element cotent is the potion's value
+	STpotion []uint
+}
+
 type State struct {
 	CurrentLevel    uint
 	HpPool          uint
 	MonsterHealth   uint
 	MonsterAttack   uint
 	AgentState      map[commons.ID]AgentState
 	InventoryMap    InventoryMap
+	PotionSlice     PotionSlice
```

Comment:
Does this slice need to exist in the global state? Since the potions are spawned and consumed immediately?

Reply:
I think it might need to be stored somwhere bc dont agents need to discuss who gets them?

Reply:
Yes, but from my understanding, when the agent view what is in the potion pool, they call the viewing function which is to read from this global state, otherwise I'm not sure where can the agent view the slice if it only lives in 'AllocateLoot', or I make it an output of 'AllocateLoot'?

Reply:
I think the alternative is to strap it into the method itself, but maybe this is just cleaner, the one thing I would note is that we can clear the slice immediately after allocation to avoid holding this memory in scope the entire time.

Reply:
Shall we do that in a new refactoring issue together with removing the slice from the global state and adding viewing to the potion pool?